### PR TITLE
add column type check when create bitmap index using create index clause

### PR DESCRIPTION
**Pull request by yangzhg**
1. add column AggregateType check when create bitmap index using create index clause
2. support more data type  in bitmap index

### CODE REVIEW

File: docs/documentation/en/administrator-guide/alter-table/alter-table-bitmap-index_EN.md [hunk 1]
```diff
@@ -56,7 +56,10 @@ Please refer to [Scheam Change](alter-table-schema-change_EN.md#Cancel Job)
 ## Notice
 * Currently only index of bitmap type is supported.
 * The bitmap index is only created on a single column.
-* Bitmap indexes can be applied to all columns of the `Duplicate` data model and key columns of the `Aggregate` and `Uniq` models.
+* Bitmap indexes can be applied to all columns of the `Duplicate` data model and key columns or `REPLACE` value columns
+ of the
+ `Aggregate` and
+ `Uniq` models.
```
Comment (wutiangan) [inline docs/documentation/en/administrator-guide/alter-table/alter-table-bitmap-index_EN.md]: change many lines to on line.

File: fe/src/main/java/org/apache/doris/analysis/CreateTableStmt.java [hunk 2]
```diff
@@ -412,9 +412,13 @@ public void analyze(Analyzer analyzer) throws UserException {
 
                             // key columns and none/replace aggregate non-key columns support
                             if (indexDef.getIndexType() == IndexDef.IndexType.BITMAP) {
-                                    if (!(colType == PrimitiveType.TINYINT || colType == PrimitiveType.SMALLINT
-                                                  || colType == PrimitiveType.INT || colType == PrimitiveType.BIGINT ||
-                                                  colType == PrimitiveType.CHAR || colType == PrimitiveType.VARCHAR)) {
+                                    if (!(colType == PrimitiveType.TINYINT || colType == PrimitiveType.SMALLINT ||
```
Comment (wutiangan) [inline fe/src/main/java/org/apache/doris/analysis/CreateTableStmt.java]: use isNumericType(), not list all type

File: fe/src/main/java/org/apache/doris/alter/Alter.java [hunk 3]
```diff
@@ -200,12 +202,39 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
...
-                                throw new AnalysisException("index for columns (" + String
-                                        .join(",", indexDef.getColumns()) + " ) already exist.");
+                        if (newColset.equals(idxSet)) {
+                            throw new AnalysisException("index for columns (" + String
+                                    .join(",", indexDef.getColumns()) + " ) already exist.");
+                        }
+                    }
+                    OlapTable olapTable = (OlapTable) table;
+                    for (String col : indexDef.getColumns()) {
+                        for (Column c : olapTable.getFullSchema()) {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/alter/Alter.java]: You can just use `olapTable.getColumn()` to get the column, no need to iterate all columns

Comment (morningman) [reply fe/src/main/java/org/apache/doris/alter/Alter.java]: And the following type checking logic looks same as in `CreateTableStmt`.
So I think they can be extract to a method?

**Review by morningman — CHANGES_REQUESTED** [review]

File: fe/src/main/java/org/apache/doris/analysis/CreateTableStmt.java (hunk 2 above)
Comment (yangzhg) [reply fe/src/main/java/org/apache/doris/analysis/CreateTableStmt.java]: it dose not support double and float and double， so cannot use  isNumericType()

File: fe/src/main/java/org/apache/doris/alter/Alter.java (hunk 3 above)
Comment (yangzhg) [reply fe/src/main/java/org/apache/doris/alter/Alter.java]: the colum name from  create index clause  may not the same case with `olapTable.getColumn ()` so i must use case `equalsIgnoreCase` to find

File: fe/src/main/java/org/apache/doris/analysis/IndexDef.java [hunk 4]
```diff
@@ -118,4 +121,32 @@ public String getComment() {
...
+            PrimitiveType colType = column.getDataType();
+            if (!(colType.isDateType() || colType.isDecimalType() || colType.isFixedPointType() ||
+                          colType.isStringType() || colType == PrimitiveType.BOOLEAN)) {
+                throw new AnalysisException(colType + " is not supported in bitmap index. "
+                        + "invalid column: " + indexColName);
+            } else if (!(column.isKey()
+                                 || column.getAggregationType() == AggregateType.NONE
+                                 || column.getAggregationType() == AggregateType.REPLACE
+                                 || column.getAggregationType() == AggregateType.REPLACE_IF_NOT_NULL)){
+                // althrough the implemention supports bf for replace non-key column,
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/analysis/IndexDef.java]: `althrough the implemention supports bf for replace non-key column`

What's meaning of `bf`?  This comment seems conflict with `bitmap 索引能够应用在 `Duplicate` 数据模型的所有列和 `Aggregate`, `Uniq` 模型的key列上和 `REPLACE` 的value列上。`

Comment (yangzhg) [reply fe/src/main/java/org/apache/doris/analysis/IndexDef.java]: its a typo  i will remove it

File: fe/src/main/java/org/apache/doris/alter/Alter.java (hunk 3 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/alter/Alter.java]: This method is case insensitive `olapTable.getColumn()`

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**